### PR TITLE
Introduce v1beta2 plugin args

### PR DESCRIPTION
**Pull request by Huang-Wei**
This PR does the following changes:

- copy v1beta1 plugin args to v1beta2 plugin args
- remove `kubeConfigPath` in all plugin args - it's not needed anymore as we're vendoring k/k 1.22.
    - NOTE: this implies old scheduler config (`CapacitySchedulingArgs`) that with _bare_ `kubeConfigPath` is totally removed. So upon migration the scheduler config.
- updated tests and cleaned up utility code
- updated the documents

### CODE REVIEW

**Comment by Huang-Wei** [comment]
@denkensk could you help review?

File: manifests/noderesourcetopology/scheduler-config.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: kubescheduler.config.k8s.io/v1beta1
+apiVersion: kubescheduler.config.k8s.io/v1beta2
```
Comment (denkensk) [inline manifests/noderesourcetopology/scheduler-config.yaml:1]: The pluginConfig for `NodeResourceTopologyMatch' also needs to be updated to remove the kubeconfigpath.

File: manifests/noderesourcetopology/scheduler-configmap.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ metadata:
   namespace: kube-system
 data:
   scheduler-config.yaml: |
-      apiVersion: kubescheduler.config.k8s.io/v1beta1
+      apiVersion: kubescheduler.config.k8s.io/v1beta2
```
Comment (denkensk) [inline manifests/noderesourcetopology/scheduler-configmap.yaml:8]: same in this file

File: pkg/apis/config/v1beta2/defaults_test.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+				TargetUtilization:         pointer.Int64Ptr(40),
+				MetricProvider: MetricProviderSpec{
+					Type: "KubernetesMetricsServer",
+				},
+			},
+		},
+		{
+			name: "set non default TargetLoadPackingArgs",
+			config: &TargetLoadPackingArgs{
+				DefaultRequests:           v1.ResourceList{v1.ResourceCPU: resource.MustParse("100m")},
```
Comment (denkensk) [inline pkg/apis/config/v1beta2/defaults_test.go:98]: Use a different value than the default to make this test case validated？

**Review by denkensk — COMMENTED** [review]
@Huang-Wei 
Thanks so much. Some small comments.

File: pkg/apis/config/v1beta2/defaults_test.go (hunk 3 above)
Comment (Huang-Wei) [reply pkg/apis/config/v1beta2/defaults_test.go:98]: This is non-default value, isn't it? (default value is `1000m`)

File: manifests/noderesourcetopology/scheduler-configmap.yaml (hunk 2 above)
Comment (Huang-Wei) [reply manifests/noderesourcetopology/scheduler-configmap.yaml:8]: done.

File: manifests/noderesourcetopology/scheduler-config.yaml (hunk 1 above)
Comment (Huang-Wei) [reply manifests/noderesourcetopology/scheduler-config.yaml:1]: done.

File: pkg/apis/config/v1beta2/defaults_test.go (hunk 3 above)
Comment (denkensk) [reply pkg/apis/config/v1beta2/defaults_test.go:98]: Yes, I misread it before.

**Comment by denkensk** [comment]
/lgtm
/approve
Thanks  @Huang-Wei

**Pull request merged**